libraries/standard/http/src/http_client.c

```diff
@@ -29,6 +34,91 @@ static HTTPStatus_t _sendHttpBody( const HTTPTransportInterface_t * pTransport,
                                    const uint8_t * pRequestBodyBuf,
                                    size_t reqBodyBufLen );
 
+/**
+ * @brief Write header based on parameters. This method also adds a trailing "\r\n".
+ * If a trailing "\r\n" already exists in the HTTP header, this method backtracks
+ * in order to write over it and updates the length accordingly.
+ *
+ * @param pRequestHeaders Request header buffer information.
+ * @param pField The header field name to write.
+ * @param fieldLen The byte length of the header field name.
+ * @param pValue The header value to write.
+ * @param valueLen The byte length of the header field value.
+ *
+ * @return #HTTP_SUCCESS if successful. If there was insufficient memory in the
+ * application buffer, then #HTTP_INSUFFICIENT_MEMORY is returned.
+ */
+static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
+                                const char * pField,
+                                size_t fieldLen,
+                                const char * pValue,
+                                size_t valueLen );
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
+                                const char * pField,
+                                size_t fieldLen,
+                                const char * pValue,
+                                size_t valueLen )
+{
```

Comment:
All the input parameters should be validated. As this is an internal function, we can add asserts here

Reply:
I agree that assert should be to validate parameters for the internal functions, but not all parameters need validation, especially if the caller has already validated them.